src/charm.py

```diff
@@ -44,12 +46,20 @@ def __init__(self, *args: typing.Any) -> None:
         except CharmConfigInvalidError as exc:
             self._update_app_and_unit_status(ops.BlockedStatus(exc.msg))
             return
-        self._webserver = GunicornWebserver(
+
+        self._database_migration = DatabaseMigration(
+            flask_container=self.unit.get_container(FLASK_CONTAINER_NAME),
```

Comment:
I think I didn't find where is checked if container can_connect. Isn't that needed?